Changes from upstream to enable build & run parser test

Can be dropped if we upgrade to 2.0.0 or later.

--- libical-1.0.1/src/test/CMakeLists.txt
+++ libical-1.0.1/src/test/CMakeLists.txt
@@ -28,9 +28,15 @@
   endif()
 endmacro()
 
-macro(testme _name _srcs)
+#macro to build (but do not create a test)
+macro(buildme _name _srcs)
   add_executable(${_name} ${_srcs})
+  add_dependencies(${_name} ical icalss icalvcal)
   target_link_libraries(${_name} ical icalss icalvcal)
+endmacro()
+
+macro(testme _name _srcs)
+  buildme(${_name} "${_srcs}")
   add_test(NAME ${_name} COMMAND ${_name})
   setprops(${_name})
 endmacro()
@@ -56,9 +62,15 @@
 
 ########### next target ###############
 
-#not sure how to use this test
-#set(parser_SRCS icaltestparser.c)
-#testme(parser "${parser_SRCS}")
+set(parser_SRCS icaltestparser.c)
+buildme(parser "${parser_SRCS}")
+
+file(GLOB TEST_FILES ${CMAKE_SOURCE_DIR}/test-data/*.ics)
+foreach(TEST_FILE ${TEST_FILES})
+  get_filename_component(TEST_NAME ${TEST_FILE} NAME_WE)
+  add_test(NAME parser-${TEST_NAME} COMMAND parser ${TEST_FILE})
+  setprops(parser-${TEST_NAME})
+endforeach()
 
 ########### next target ###############
 
